### DISCUSSION TITLE
Public Class attributes in preload script

### DISCUSSION THREAD

**Discussion by Dragony** [nklayman/vue-cli-plugin-electron-builder#1490] [Q&A] [answered]
Hello there!

I'm trying to use [Public Class attributes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Public_class_fields) in a preload.js script. When I run `npm run electron:serve` I get the following Error:

```
Module parse failed: Unexpected token (8:13)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|
| class Store {
>     filename = isDevelopment ? '/config_dev.json' : '/config.json';
|     storage = {};
|     configPath = null;
```

This seems to indicate that this javascript feature is not supported. It's strange, because I can use this feature anywhere else in the app. In the frontend app via webpack/babel and in my background script it's supported via nodejs. How can I enable this feature for my preload script?

For the record I've registed the script as follows:

```
pluginOptions: {
    electronBuilder: {
        preload: {
            preload: 'src/preload.js'
        }
    }
}
```

**Comment by nklayman** [comment, marked as answer]
This is because the preload.js file isn't transpiled with babel by default. You need to enable it for the main process, see https://github.com/nklayman/vue-cli-plugin-electron-builder/issues/204#issuecomment-466122865 for details. It's for class properties specifically but it should be simple to get it to work for your usecase.
